src/tools/tidy/src/alphabetical.rs

```diff
@@ -71,7 +76,7 @@ fn check_section<'a>(
         let trimmed_line = line.trim_start_matches(' ');
 
         if trimmed_line.starts_with("//")
-            || trimmed_line.starts_with("#[")
+            || trimmed_line.starts_with("#")
```

Comment:
We use this to check that feature attributes are sorted. Will this still work?

Reply:
Good point. As written, no, but it should be possible to tweak the string matching so it does continue to work.

Reply:
Wait a minute, feature attributes aren't sorted. E.g. `compiler/rustc_arena/lib.rs` is an example without sorting. And this tidy sorting is only between `tidy-alphabetical-{start,end}` markers.

Reply:
This case for instance: https://github.com/rust-lang/rust/blob/master/library/core/src/lib.rs#L113-L195

Reply:
This is supposed to make it ignore attributes in the features macro. Extending it to cover `#` (toml comments) sounds fine, but should have a comment that it also ignored attributes.

Reply:
I have updated the code to ignore lines starting with `#` except for lines starting with `#!`.

Reply:
Note: thanks to #117149 landing, we now have unit tests that would have caught this problem.